### PR TITLE
MTDSA 2708: Added tradingAllowance to self-employment annual summaries

### PR DESCRIPTION
**Pull request by MeliaDavies**
Please check if this change includes (or even requires) the following:

 - [ ] unit tests
 - [ ] functional tests
 - [ ] audit events
 - [ ] RAML documentation
 - [ ] logging

### CODE REVIEW

File: test/uk/gov/hmrc/r2/selfassessmentapi/models/selfemployment/SelfEmploymentAnnualSummarySpec.scala [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+      assertValidationPasses(summary)
+    }
+  }
+
+
+  "from" should {
+    "correctly map a DES self-employment to an API self-employment" in {
+      val desSelfEmployment = des.selfemployment.SelfEmploymentAnnualSummary(
+        annualAdjustments = Some(des.selfemployment.AnnualAdjustments(
```
Comment (london-coder) [inline test/uk/gov/hmrc/r2/selfassessmentapi/models/selfemployment/SelfEmploymentAnnualSummarySpec.scala]: This is like an assert(2 == 2), does it exercise code?

File: app/uk/gov/hmrc/r2/selfassessmentapi/models/des/selfemployment/SelfEmployment.scala [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  implicit val reads: Reads[SelfEmployment] = (
+    (__ \ "incomeSourceId").readNullable[String] and
+      (__ \ "accountingPeriodStartDate").read[String] and
+      (__ \ "accountingPeriodEndDate").read[String] and
+      (__ \ "tradingName").read[String] and
+      (__ \ "businessAddressDetails").readNullable[SelfEmploymentAddress] and
+      (__ \ "typeOfBusiness").readNullable[String] and
+      (__ \ "tradingStartDate").readNullable[String] and
+      (__ \ "cashOrAccruals").read[String]
+    )(SelfEmployment.apply _)
```
Comment (london-coder) [inline app/uk/gov/hmrc/r2/selfassessmentapi/models/des/selfemployment/SelfEmployment.scala]: When the JSON contains a value in businessAddressDetails, will there be a value in the case class addressDetails field?

Comment (ma3574) [reply app/uk/gov/hmrc/r2/selfassessmentapi/models/des/selfemployment/SelfEmployment.scala]: Yes, `addressDetails` will get populated by the data in the `businessAddressDetails` field on the response from DES

Comment (london-coder) [reply app/uk/gov/hmrc/r2/selfassessmentapi/models/des/selfemployment/SelfEmployment.scala]: Cool, thanks Mark, I'm going to merge this.

**Pull request merged**